BizHawk.Client.EmuHawk/Communication.cs

```diff
@@ -194,42 +193,63 @@ public void Connect()
 				soc.Connect(remoteEP);
 				connected = true;
 				soc.ReceiveTimeout = 5;
-				
 			}
-			public void SetIp(string ip_)
+			public void SetIp(string ip_, int port_)
 			{
 				ip = ip_;
+				port = port_;
 				ipAdd = System.Net.IPAddress.Parse(ip);
 				remoteEP = new IPEndPoint(ipAdd, port);
+
 			}
-			public void SetIp(string ip_, int port_)
+
+			public void SetIp(string ip_)
 			{
 				ip = ip_;
-				port = port_;
 				ipAdd = System.Net.IPAddress.Parse(ip);
-				remoteEP = new IPEndPoint(ipAdd, port);
-				
+				Connect();
+			}
+
+			public void SetPort(int port_)
+			{
+				port = port_;
+				Connect();
+			}
+
+			public int GetPort()
+			{
+				return port;
+			}
+
+			public string GetIp()

```

Comment:
There are getters and setters in C#, see [MSDN](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties). You make a "backing field" (private) and use the get/set keywords to control access or call e.g. `Connect` after assignment.

Reply:
Makes sense, will be changed accordingly.